// @wago/wdx-schema-base/build/WDX/Schema/Model/Package/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Package
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.Package = exports.Author = exports.Maintainer = exports.User = exports.DistTags = exports.Versions = exports.Version = exports.Status = exports.UninstallRequestBody = exports.InstallRequestBody = exports.ListRequestBody = exports.InfoRequestBody = void 0;
const InfoRequestBody_1 = require("./InfoRequestBody");
Object.defineProperty(exports, "InfoRequestBody", { enumerable: true, get: function () { return InfoRequestBody_1.InfoRequestBody; } });
const ListRequestBody_1 = require("./ListRequestBody");
Object.defineProperty(exports, "ListRequestBody", { enumerable: true, get: function () { return ListRequestBody_1.ListRequestBody; } });
const InstallRequestBody_1 = require("./InstallRequestBody");
Object.defineProperty(exports, "InstallRequestBody", { enumerable: true, get: function () { return InstallRequestBody_1.InstallRequestBody; } });
const UninstallRequestBody_1 = require("./UninstallRequestBody");
Object.defineProperty(exports, "UninstallRequestBody", { enumerable: true, get: function () { return UninstallRequestBody_1.UninstallRequestBody; } });
const Status_1 = require("./Status");
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return Status_1.Status; } });
const Version_1 = require("./Version");
Object.defineProperty(exports, "Version", { enumerable: true, get: function () { return Version_1.Version; } });
const Versions_1 = require("./Versions");
Object.defineProperty(exports, "Versions", { enumerable: true, get: function () { return Versions_1.Versions; } });
const DistTags_1 = require("./DistTags");
Object.defineProperty(exports, "DistTags", { enumerable: true, get: function () { return DistTags_1.DistTags; } });
const User_1 = require("./User");
Object.defineProperty(exports, "User", { enumerable: true, get: function () { return User_1.User; } });
const Maintainer_1 = require("./Maintainer");
Object.defineProperty(exports, "Maintainer", { enumerable: true, get: function () { return Maintainer_1.Maintainer; } });
const Author_1 = require("./Author");
Object.defineProperty(exports, "Author", { enumerable: true, get: function () { return Author_1.Author; } });
const Package_1 = require("./Package");
Object.defineProperty(exports, "Package", { enumerable: true, get: function () { return Package_1.Package; } });
