// @wago/wdx-schema-base/build/WDX/Schema/Model/Package/Package.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Package
 *
 * @copyright 2024 Elrest AutomationsSysteme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.Package = void 0;
const Status_1 = require("./Status");
const Versions_1 = require("./Versions");
class Package {
    constructor(id, author = new Array, name = '', description = '', documentation = '', license = '', status = Status_1.Status.Available, currentVersion = '', newestVersion = '', installed = false, mandatory = false, versions = new Versions_1.Versions) {
        this.id = id;
        this.author = author;
        this.name = name;
        this.description = description;
        this.documentation = documentation;
        this.license = license;
        this.status = status;
        this.currentVersion = currentVersion;
        this.newestVersion = newestVersion;
        this.installed = installed;
        this.mandatory = mandatory;
        this.versions = versions;
    }
}
exports.Package = Package;
