// @wago/wdx-schema-base/build/WDX/Schema/Message/Alarm/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Messages
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportHistoryResponse = exports.ExportHistoryRequest = exports.ExportResponse = exports.ExportRequest = exports.SetInactiveResponse = exports.SetInactiveRequest = exports.SetActiveResponse = exports.SetActiveRequest = exports.GetStatusResponse = exports.GetStatusRequest = exports.Update = exports.UnsubscribeResponse = exports.UnsubscribeRequest = exports.SubscribeResponse = exports.SubscribeRequest = exports.SetResponse = exports.SetRequest = exports.DeleteResponse = exports.DeleteRequest = exports.ListHistoryResponse = exports.ListHistoryRequest = exports.ListResponse = exports.ListRequest = exports.ConfirmResponse = exports.ConfirmRequest = exports.DetailResponse = exports.DetailRequest = void 0;
const ConfirmRequest_1 = require("./ConfirmRequest");
Object.defineProperty(exports, "ConfirmRequest", { enumerable: true, get: function () { return ConfirmRequest_1.ConfirmRequest; } });
const ConfirmResponse_1 = require("./ConfirmResponse");
Object.defineProperty(exports, "ConfirmResponse", { enumerable: true, get: function () { return ConfirmResponse_1.ConfirmResponse; } });
const ListRequest_1 = require("./ListRequest");
Object.defineProperty(exports, "ListRequest", { enumerable: true, get: function () { return ListRequest_1.ListRequest; } });
const ListResponse_1 = require("./ListResponse");
Object.defineProperty(exports, "ListResponse", { enumerable: true, get: function () { return ListResponse_1.ListResponse; } });
const DetailRequest_1 = require("./DetailRequest");
Object.defineProperty(exports, "DetailRequest", { enumerable: true, get: function () { return DetailRequest_1.DetailRequest; } });
const DetailResponse_1 = require("./DetailResponse");
Object.defineProperty(exports, "DetailResponse", { enumerable: true, get: function () { return DetailResponse_1.DetailResponse; } });
const ListHistoryRequest_1 = require("./ListHistoryRequest");
Object.defineProperty(exports, "ListHistoryRequest", { enumerable: true, get: function () { return ListHistoryRequest_1.ListHistoryRequest; } });
const ListHistoryResponse_1 = require("./ListHistoryResponse");
Object.defineProperty(exports, "ListHistoryResponse", { enumerable: true, get: function () { return ListHistoryResponse_1.ListHistoryResponse; } });
const DeleteRequest_1 = require("./DeleteRequest");
Object.defineProperty(exports, "DeleteRequest", { enumerable: true, get: function () { return DeleteRequest_1.DeleteRequest; } });
const DeleteResponse_1 = require("./DeleteResponse");
Object.defineProperty(exports, "DeleteResponse", { enumerable: true, get: function () { return DeleteResponse_1.DeleteResponse; } });
const SetRequest_1 = require("./SetRequest");
Object.defineProperty(exports, "SetRequest", { enumerable: true, get: function () { return SetRequest_1.SetRequest; } });
const SetResponse_1 = require("./SetResponse");
Object.defineProperty(exports, "SetResponse", { enumerable: true, get: function () { return SetResponse_1.SetResponse; } });
const Update_1 = require("./Update");
Object.defineProperty(exports, "Update", { enumerable: true, get: function () { return Update_1.Update; } });
const SubscribeRequest_1 = require("./SubscribeRequest");
Object.defineProperty(exports, "SubscribeRequest", { enumerable: true, get: function () { return SubscribeRequest_1.SubscribeRequest; } });
const SubscribeResponse_1 = require("./SubscribeResponse");
Object.defineProperty(exports, "SubscribeResponse", { enumerable: true, get: function () { return SubscribeResponse_1.SubscribeResponse; } });
const UnsubscribeRequest_1 = require("./UnsubscribeRequest");
Object.defineProperty(exports, "UnsubscribeRequest", { enumerable: true, get: function () { return UnsubscribeRequest_1.UnsubscribeRequest; } });
const UnsubscribeResponse_1 = require("./UnsubscribeResponse");
Object.defineProperty(exports, "UnsubscribeResponse", { enumerable: true, get: function () { return UnsubscribeResponse_1.UnsubscribeResponse; } });
const GetStatusRequest_1 = require("./GetStatusRequest");
Object.defineProperty(exports, "GetStatusRequest", { enumerable: true, get: function () { return GetStatusRequest_1.GetStatusRequest; } });
const GetStatusResponse_1 = require("./GetStatusResponse");
Object.defineProperty(exports, "GetStatusResponse", { enumerable: true, get: function () { return GetStatusResponse_1.GetStatusResponse; } });
const SetActiveRequest_1 = require("./SetActiveRequest");
Object.defineProperty(exports, "SetActiveRequest", { enumerable: true, get: function () { return SetActiveRequest_1.SetActiveRequest; } });
const SetActiveResponse_1 = require("./SetActiveResponse");
Object.defineProperty(exports, "SetActiveResponse", { enumerable: true, get: function () { return SetActiveResponse_1.SetActiveResponse; } });
const SetInactiveRequest_1 = require("./SetInactiveRequest");
Object.defineProperty(exports, "SetInactiveRequest", { enumerable: true, get: function () { return SetInactiveRequest_1.SetInactiveRequest; } });
const SetInactiveResponse_1 = require("./SetInactiveResponse");
Object.defineProperty(exports, "SetInactiveResponse", { enumerable: true, get: function () { return SetInactiveResponse_1.SetInactiveResponse; } });
const ExportRequest_1 = require("./ExportRequest");
Object.defineProperty(exports, "ExportRequest", { enumerable: true, get: function () { return ExportRequest_1.ExportRequest; } });
const ExportResponse_1 = require("./ExportResponse");
Object.defineProperty(exports, "ExportResponse", { enumerable: true, get: function () { return ExportResponse_1.ExportResponse; } });
const ExportHistoryRequest_1 = require("./ExportHistoryRequest");
Object.defineProperty(exports, "ExportHistoryRequest", { enumerable: true, get: function () { return ExportHistoryRequest_1.ExportHistoryRequest; } });
const ExportHistoryResponse_1 = require("./ExportHistoryResponse");
Object.defineProperty(exports, "ExportHistoryResponse", { enumerable: true, get: function () { return ExportHistoryResponse_1.ExportHistoryResponse; } });
