// @wago/wdx-schema-base/build/WDX/Schema/Model/Exception/index.js
/**
 * Elrest eDesign - Runtime - Library - Package
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnexpectedErrorException = exports.WrongRequestException = exports.ForbiddenException = exports.NotFoundException = exports.AbstractException = exports.Code = void 0;
const Code_1 = require("./Code");
Object.defineProperty(exports, "Code", { enumerable: true, get: function () { return Code_1.Code; } });
const AbstractException_1 = require("./AbstractException");
Object.defineProperty(exports, "AbstractException", { enumerable: true, get: function () { return AbstractException_1.AbstractException; } });
const NotFoundException_1 = require("./NotFoundException");
Object.defineProperty(exports, "NotFoundException", { enumerable: true, get: function () { return NotFoundException_1.NotFoundException; } });
const ForbiddenException_1 = require("./ForbiddenException");
Object.defineProperty(exports, "ForbiddenException", { enumerable: true, get: function () { return ForbiddenException_1.ForbiddenException; } });
const WrongRequestException_1 = require("./WrongRequestException");
Object.defineProperty(exports, "WrongRequestException", { enumerable: true, get: function () { return WrongRequestException_1.WrongRequestException; } });
const UnexpectedErrorException_1 = require("./UnexpectedErrorException");
Object.defineProperty(exports, "UnexpectedErrorException", { enumerable: true, get: function () { return UnexpectedErrorException_1.UnexpectedErrorException; } });
