// @wago/wdx-schema-base/build/WDX/Schema/Message/Instance/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Messages Instance
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.WhoIsResponse = exports.WhoIsRequest = exports.SaveResponse = exports.SaveRequest = exports.StopResponse = exports.StopRequest = exports.DeleteResponse = exports.DeleteRequest = exports.StartResponse = exports.StartRequest = exports.RestartResponse = exports.RestartRequest = exports.LogUnsubscribeResponseMessage = exports.LogUnsubscribeRequestMessage = exports.LogSubscribeResponseMessage = exports.LogSubscribeRequestMessage = exports.LogMessage = exports.ListNamespacesResponse = exports.ListNamespacesRequest = exports.ListLogResponse = exports.ListLogRequest = exports.ListResponse = exports.ListRequest = exports.DetailResponse = exports.DetailRequest = exports.IAmGone = exports.IAmError = exports.IAm = exports.UnsubscribeResponse = exports.UnsubscribeRequest = exports.SubscribeResponse = exports.SubscribeRequest = exports.Monitor = void 0;
const IAm_1 = require("./IAm");
Object.defineProperty(exports, "IAm", { enumerable: true, get: function () { return IAm_1.IAm; } });
const IAmError_1 = require("./IAmError");
Object.defineProperty(exports, "IAmError", { enumerable: true, get: function () { return IAmError_1.IAmError; } });
const IAmGone_1 = require("./IAmGone");
Object.defineProperty(exports, "IAmGone", { enumerable: true, get: function () { return IAmGone_1.IAmGone; } });
const DetailRequest_1 = require("./DetailRequest");
Object.defineProperty(exports, "DetailRequest", { enumerable: true, get: function () { return DetailRequest_1.DetailRequest; } });
const DetailResponse_1 = require("./DetailResponse");
Object.defineProperty(exports, "DetailResponse", { enumerable: true, get: function () { return DetailResponse_1.DetailResponse; } });
const ListRequest_1 = require("./ListRequest");
Object.defineProperty(exports, "ListRequest", { enumerable: true, get: function () { return ListRequest_1.ListRequest; } });
const ListResponse_1 = require("./ListResponse");
Object.defineProperty(exports, "ListResponse", { enumerable: true, get: function () { return ListResponse_1.ListResponse; } });
const ListLogRequest_1 = require("./ListLogRequest");
Object.defineProperty(exports, "ListLogRequest", { enumerable: true, get: function () { return ListLogRequest_1.ListLogRequest; } });
const ListLogResponse_1 = require("./ListLogResponse");
Object.defineProperty(exports, "ListLogResponse", { enumerable: true, get: function () { return ListLogResponse_1.ListLogResponse; } });
const ListNamespacesRequest_1 = require("./ListNamespacesRequest");
Object.defineProperty(exports, "ListNamespacesRequest", { enumerable: true, get: function () { return ListNamespacesRequest_1.ListNamespacesRequest; } });
const ListNamespacesResponse_1 = require("./ListNamespacesResponse");
Object.defineProperty(exports, "ListNamespacesResponse", { enumerable: true, get: function () { return ListNamespacesResponse_1.ListNamespacesResponse; } });
const LogMessage_1 = require("./LogMessage");
Object.defineProperty(exports, "LogMessage", { enumerable: true, get: function () { return LogMessage_1.LogMessage; } });
const LogSubscribeRequestMessage_1 = require("./LogSubscribeRequestMessage");
Object.defineProperty(exports, "LogSubscribeRequestMessage", { enumerable: true, get: function () { return LogSubscribeRequestMessage_1.LogSubscribeRequestMessage; } });
const LogSubscribeResponseMessage_1 = require("./LogSubscribeResponseMessage");
Object.defineProperty(exports, "LogSubscribeResponseMessage", { enumerable: true, get: function () { return LogSubscribeResponseMessage_1.LogSubscribeResponseMessage; } });
const LogUnsubscribeRequestMessage_1 = require("./LogUnsubscribeRequestMessage");
Object.defineProperty(exports, "LogUnsubscribeRequestMessage", { enumerable: true, get: function () { return LogUnsubscribeRequestMessage_1.LogUnsubscribeRequestMessage; } });
const LogUnsubscribeResponseMessage_1 = require("./LogUnsubscribeResponseMessage");
Object.defineProperty(exports, "LogUnsubscribeResponseMessage", { enumerable: true, get: function () { return LogUnsubscribeResponseMessage_1.LogUnsubscribeResponseMessage; } });
const Monitor_1 = require("./Monitor");
Object.defineProperty(exports, "Monitor", { enumerable: true, get: function () { return Monitor_1.Monitor; } });
const SubscribeRequest_1 = require("./SubscribeRequest");
Object.defineProperty(exports, "SubscribeRequest", { enumerable: true, get: function () { return SubscribeRequest_1.SubscribeRequest; } });
const SubscribeResponse_1 = require("./SubscribeResponse");
Object.defineProperty(exports, "SubscribeResponse", { enumerable: true, get: function () { return SubscribeResponse_1.SubscribeResponse; } });
const UnsubscribeRequest_1 = require("./UnsubscribeRequest");
Object.defineProperty(exports, "UnsubscribeRequest", { enumerable: true, get: function () { return UnsubscribeRequest_1.UnsubscribeRequest; } });
const UnsubscribeResponse_1 = require("./UnsubscribeResponse");
Object.defineProperty(exports, "UnsubscribeResponse", { enumerable: true, get: function () { return UnsubscribeResponse_1.UnsubscribeResponse; } });
const RestartRequest_1 = require("./RestartRequest");
Object.defineProperty(exports, "RestartRequest", { enumerable: true, get: function () { return RestartRequest_1.RestartRequest; } });
const RestartResponse_1 = require("./RestartResponse");
Object.defineProperty(exports, "RestartResponse", { enumerable: true, get: function () { return RestartResponse_1.RestartResponse; } });
const StartRequest_1 = require("./StartRequest");
Object.defineProperty(exports, "StartRequest", { enumerable: true, get: function () { return StartRequest_1.StartRequest; } });
const StartResponse_1 = require("./StartResponse");
Object.defineProperty(exports, "StartResponse", { enumerable: true, get: function () { return StartResponse_1.StartResponse; } });
const DeleteRequest_1 = require("./DeleteRequest");
Object.defineProperty(exports, "DeleteRequest", { enumerable: true, get: function () { return DeleteRequest_1.DeleteRequest; } });
const DeleteResponse_1 = require("./DeleteResponse");
Object.defineProperty(exports, "DeleteResponse", { enumerable: true, get: function () { return DeleteResponse_1.DeleteResponse; } });
const StopRequest_1 = require("./StopRequest");
Object.defineProperty(exports, "StopRequest", { enumerable: true, get: function () { return StopRequest_1.StopRequest; } });
const StopResponse_1 = require("./StopResponse");
Object.defineProperty(exports, "StopResponse", { enumerable: true, get: function () { return StopResponse_1.StopResponse; } });
const SaveRequest_1 = require("./SaveRequest");
Object.defineProperty(exports, "SaveRequest", { enumerable: true, get: function () { return SaveRequest_1.SaveRequest; } });
const SaveResponse_1 = require("./SaveResponse");
Object.defineProperty(exports, "SaveResponse", { enumerable: true, get: function () { return SaveResponse_1.SaveResponse; } });
const WhoIsRequest_1 = require("./WhoIsRequest");
Object.defineProperty(exports, "WhoIsRequest", { enumerable: true, get: function () { return WhoIsRequest_1.WhoIsRequest; } });
const WhoIsResponse_1 = require("./WhoIsResponse");
Object.defineProperty(exports, "WhoIsResponse", { enumerable: true, get: function () { return WhoIsResponse_1.WhoIsResponse; } });
