// @wago/wdx-schema-base/build/WDX/Schema/Model/Alarm/AlarmUpdate.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Data Schema
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlarmUpdate = void 0;
const AlarmCondition_1 = require("./AlarmCondition");
class AlarmUpdate {
    constructor(id, name, active = true, message, messageOff, number, type, status, cause, reaction, correction, createDateTime = Date.now(), updatedDateTime = createDateTime, conditions = [
        new AlarmCondition_1.AlarmCondition(),
    ]) {
        this.id = id;
        this.name = name;
        this.active = active;
        this.message = message;
        this.messageOff = messageOff;
        this.number = number;
        this.type = type;
        this.status = status;
        this.cause = cause;
        this.reaction = reaction;
        this.correction = correction;
        this.createDateTime = createDateTime;
        this.updatedDateTime = updatedDateTime;
        this.conditions = conditions;
    }
}
exports.AlarmUpdate = AlarmUpdate;
