// @wago/wdx-schema-base/build/WDX/Schema/Model/Alarm/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Alarm
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlarmConditionExpression = exports.AlarmType = exports.AlarmCondition = exports.AlarmStatus = exports.AlarmHistory = exports.AlarmUpdate = exports.AlarmMessage = exports.Alarm = exports.AlarmClassification = void 0;
const Alarm_1 = require("./Alarm");
Object.defineProperty(exports, "Alarm", { enumerable: true, get: function () { return Alarm_1.Alarm; } });
const AlarmClassification_1 = require("./AlarmClassification");
Object.defineProperty(exports, "AlarmClassification", { enumerable: true, get: function () { return AlarmClassification_1.AlarmClassification; } });
const AlarmHistory_1 = require("./AlarmHistory");
Object.defineProperty(exports, "AlarmHistory", { enumerable: true, get: function () { return AlarmHistory_1.AlarmHistory; } });
const AlarmStatus_1 = require("./AlarmStatus");
Object.defineProperty(exports, "AlarmStatus", { enumerable: true, get: function () { return AlarmStatus_1.AlarmStatus; } });
const AlarmType_1 = require("./AlarmType");
Object.defineProperty(exports, "AlarmType", { enumerable: true, get: function () { return AlarmType_1.AlarmType; } });
const AlarmCondition_1 = require("./AlarmCondition");
Object.defineProperty(exports, "AlarmCondition", { enumerable: true, get: function () { return AlarmCondition_1.AlarmCondition; } });
const AlarmConditionExpression_1 = require("./AlarmConditionExpression");
Object.defineProperty(exports, "AlarmConditionExpression", { enumerable: true, get: function () { return AlarmConditionExpression_1.AlarmConditionExpression; } });
const AlarmUpdate_1 = require("./AlarmUpdate");
Object.defineProperty(exports, "AlarmUpdate", { enumerable: true, get: function () { return AlarmUpdate_1.AlarmUpdate; } });
const AlarmMessage_1 = require("./AlarmMessage");
Object.defineProperty(exports, "AlarmMessage", { enumerable: true, get: function () { return AlarmMessage_1.AlarmMessage; } });
