// @wago/wdx-schema-base/build/WDX/Schema/Model/Trend/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Trend
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FollowType = exports.Type = exports.InterpolationType = exports.Interpolation = exports.ExportRequestBody = exports.DefaultFormat = exports.DataSet = exports.Trend = exports.Export = exports.DataRequestBody = exports.PoolPeriod = exports.Data = void 0;
const Trend_1 = require("./Trend");
Object.defineProperty(exports, "Trend", { enumerable: true, get: function () { return Trend_1.Trend; } });
const PoolPeriod_1 = require("./PoolPeriod");
Object.defineProperty(exports, "PoolPeriod", { enumerable: true, get: function () { return PoolPeriod_1.PoolPeriod; } });
const DataSet_1 = require("./DataSet");
Object.defineProperty(exports, "DataSet", { enumerable: true, get: function () { return DataSet_1.DataSet; } });
const DefaultFormat_1 = require("./DefaultFormat");
Object.defineProperty(exports, "DefaultFormat", { enumerable: true, get: function () { return DefaultFormat_1.DefaultFormat; } });
const Data_1 = require("./Data");
Object.defineProperty(exports, "Data", { enumerable: true, get: function () { return Data_1.Data; } });
const DataRequestBody_1 = require("./DataRequestBody");
Object.defineProperty(exports, "DataRequestBody", { enumerable: true, get: function () { return DataRequestBody_1.DataRequestBody; } });
const Export = __importStar(require("./Export"));
exports.Export = Export;
const ExportRequestBody_1 = require("./ExportRequestBody");
Object.defineProperty(exports, "ExportRequestBody", { enumerable: true, get: function () { return ExportRequestBody_1.ExportRequestBody; } });
const Interpolation_1 = require("./Interpolation");
Object.defineProperty(exports, "Interpolation", { enumerable: true, get: function () { return Interpolation_1.Interpolation; } });
const InterpolationType_1 = require("./InterpolationType");
Object.defineProperty(exports, "InterpolationType", { enumerable: true, get: function () { return InterpolationType_1.InterpolationType; } });
const Type_1 = require("./Type");
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return Type_1.Type; } });
const FollowType_1 = require("./FollowType");
Object.defineProperty(exports, "FollowType", { enumerable: true, get: function () { return FollowType_1.FollowType; } });
