// @wago/wdx-schema-base/build/WDX/Schema/Message/Access/User/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Access User Package
 *
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteResponse = exports.DeleteRequest = exports.SaveResponse = exports.SaveRequest = exports.ListResponse = exports.ListRequest = void 0;
const ListRequest_1 = require("./ListRequest");
Object.defineProperty(exports, "ListRequest", { enumerable: true, get: function () { return ListRequest_1.ListRequest; } });
const ListResponse_1 = require("./ListResponse");
Object.defineProperty(exports, "ListResponse", { enumerable: true, get: function () { return ListResponse_1.ListResponse; } });
const SaveRequest_1 = require("./SaveRequest");
Object.defineProperty(exports, "SaveRequest", { enumerable: true, get: function () { return SaveRequest_1.SaveRequest; } });
const SaveResponse_1 = require("./SaveResponse");
Object.defineProperty(exports, "SaveResponse", { enumerable: true, get: function () { return SaveResponse_1.SaveResponse; } });
const DeleteRequest_1 = require("./DeleteRequest");
Object.defineProperty(exports, "DeleteRequest", { enumerable: true, get: function () { return DeleteRequest_1.DeleteRequest; } });
const DeleteResponse_1 = require("./DeleteResponse");
Object.defineProperty(exports, "DeleteResponse", { enumerable: true, get: function () { return DeleteResponse_1.DeleteResponse; } });
