// @wago/wdx-schema-base/build/WDX/Schema/Model/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Package
 *
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Translation = exports.Chart = exports.Trend = exports.Alarm = exports.Script = exports.Runtime = exports.Repository = exports.Instance = exports.Package = exports.Host = exports.Filesystem = exports.Exception = exports.Device = exports.Data = exports.Pagination = exports.Access = void 0;
const Data = __importStar(require("./Data"));
exports.Data = Data;
const Device = __importStar(require("./Device"));
exports.Device = Device;
const Exception = __importStar(require("./Exception"));
exports.Exception = Exception;
const Pagination = __importStar(require("./Pagination"));
exports.Pagination = Pagination;
const Filesystem = __importStar(require("./Filesystem"));
exports.Filesystem = Filesystem;
const Host = __importStar(require("./Host"));
exports.Host = Host;
const Instance = __importStar(require("./Instance"));
exports.Instance = Instance;
const Package = __importStar(require("./Package"));
exports.Package = Package;
const Repository = __importStar(require("./Repository"));
exports.Repository = Repository;
const Runtime = __importStar(require("./Runtime"));
exports.Runtime = Runtime;
const Script = __importStar(require("./Script"));
exports.Script = Script;
const Alarm = __importStar(require("./Alarm"));
exports.Alarm = Alarm;
const Trend = __importStar(require("./Trend"));
exports.Trend = Trend;
const Chart = __importStar(require("./Chart"));
exports.Chart = Chart;
const Access = __importStar(require("./Access"));
exports.Access = Access;
const Translation = __importStar(require("./Translation"));
exports.Translation = Translation;
