// @wago/wdx-schema-base/build/WDX/Schema/Message/Runtime/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Messages Runtime
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogrotateResponse = exports.LogrotateRequest = exports.RestartResponse = exports.RestartRequest = exports.StopResponse = exports.StopRequest = exports.StartResponse = exports.StartRequest = exports.MonitorUnsubscribeResponse = exports.MonitorUnsubscribeRequest = exports.MonitorSubscribeResponse = exports.MonitorSubscribeRequest = exports.Monitor = exports.InfoResponse = exports.InfoRequest = void 0;
const InfoRequest_1 = require("./InfoRequest");
Object.defineProperty(exports, "InfoRequest", { enumerable: true, get: function () { return InfoRequest_1.InfoRequest; } });
const InfoResponse_1 = require("./InfoResponse");
Object.defineProperty(exports, "InfoResponse", { enumerable: true, get: function () { return InfoResponse_1.InfoResponse; } });
const Monitor_1 = require("./Monitor");
Object.defineProperty(exports, "Monitor", { enumerable: true, get: function () { return Monitor_1.Monitor; } });
const LogrotateRequest_1 = require("./LogrotateRequest");
Object.defineProperty(exports, "LogrotateRequest", { enumerable: true, get: function () { return LogrotateRequest_1.LogrotateRequest; } });
const LogrotateResponse_1 = require("./LogrotateResponse");
Object.defineProperty(exports, "LogrotateResponse", { enumerable: true, get: function () { return LogrotateResponse_1.LogrotateResponse; } });
const MonitorSubscribeRequest_1 = require("./MonitorSubscribeRequest");
Object.defineProperty(exports, "MonitorSubscribeRequest", { enumerable: true, get: function () { return MonitorSubscribeRequest_1.MonitorSubscribeRequest; } });
const MonitorSubscribeResponse_1 = require("./MonitorSubscribeResponse");
Object.defineProperty(exports, "MonitorSubscribeResponse", { enumerable: true, get: function () { return MonitorSubscribeResponse_1.MonitorSubscribeResponse; } });
const MonitorUnsubscribeRequest_1 = require("./MonitorUnsubscribeRequest");
Object.defineProperty(exports, "MonitorUnsubscribeRequest", { enumerable: true, get: function () { return MonitorUnsubscribeRequest_1.MonitorUnsubscribeRequest; } });
const MonitorUnsubscribeResponse_1 = require("./MonitorUnsubscribeResponse");
Object.defineProperty(exports, "MonitorUnsubscribeResponse", { enumerable: true, get: function () { return MonitorUnsubscribeResponse_1.MonitorUnsubscribeResponse; } });
const StartRequest_1 = require("./StartRequest");
Object.defineProperty(exports, "StartRequest", { enumerable: true, get: function () { return StartRequest_1.StartRequest; } });
const StartResponse_1 = require("./StartResponse");
Object.defineProperty(exports, "StartResponse", { enumerable: true, get: function () { return StartResponse_1.StartResponse; } });
const StopRequest_1 = require("./StopRequest");
Object.defineProperty(exports, "StopRequest", { enumerable: true, get: function () { return StopRequest_1.StopRequest; } });
const StopResponse_1 = require("./StopResponse");
Object.defineProperty(exports, "StopResponse", { enumerable: true, get: function () { return StopResponse_1.StopResponse; } });
const RestartRequest_1 = require("./RestartRequest");
Object.defineProperty(exports, "RestartRequest", { enumerable: true, get: function () { return RestartRequest_1.RestartRequest; } });
const RestartResponse_1 = require("./RestartResponse");
Object.defineProperty(exports, "RestartResponse", { enumerable: true, get: function () { return RestartResponse_1.RestartResponse; } });
