// @wago/wdx-schema-base/build/WDX/Schema/Model/Chart/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Chart Package
 *
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TooltipMode = exports.PointStyle = exports.Tooltip = exports.LegendLabel = exports.LegendPosition = exports.Legend = exports.Layout = exports.Font = exports.Padding = exports.Title = exports.TitlePosition = exports.TextAlign = exports.HoriziontalAlign = exports.ChartDataSet = exports.AxisTickMajor = exports.AxisTick = exports.AxisGrid = exports.AxisBorder = exports.AxisPosition = exports.AxisBounds = exports.AxisType = exports.AxisClassification = exports.AxisTimeOptions = exports.Axis = exports.Clip = exports.Label = exports.DataRequestBody = exports.Export = exports.SourceType = exports.Type = exports.ExportRequestBody = exports.ParsingOptionScale = exports.ParsingOption = exports.RegressionLine = exports.Regression = exports.RegressionType = exports.ConfigurationRequestBody = exports.StreamingPlugin = exports.ZoomPluginScaleLimits = exports.ZoomPluginZoomPinch = exports.ZoomPluginZoomDrag = exports.ZoomPluginZoomWheel = exports.ZoomPluginZoom = exports.ZoomPluginOverScaleMode = exports.ZoomPluginScaleMode = exports.ZoomPluginModifierKey = exports.ZoomPluginMode = exports.ZoomPluginPan = exports.ZoomPluginDrawTime = exports.ZoomPlugin = void 0;
exports.AxisTimeDisplayFormats = exports.TimeUnit = exports.Intervals = exports.IntervalPickerPlugin = exports.AxisTime = exports.TickSource = exports.ConfigurationOptions = exports.ConfigurationDataSet = exports.ConfigurationData = exports.Configuration = exports.Chart = exports.AxisRadialPointLabel = exports.AxisRadialAngleLine = exports.AxisRadial = exports.AxisCartesian = exports.AxisTypeRadial = exports.FontStyle = exports.FontWeight = exports.TooltipPositionMode = void 0;
const Export = __importStar(require("./Export"));
exports.Export = Export;
const StreamingPlugin_1 = require("./StreamingPlugin");
Object.defineProperty(exports, "StreamingPlugin", { enumerable: true, get: function () { return StreamingPlugin_1.StreamingPlugin; } });
const ConfigurationRequestBody_1 = require("./ConfigurationRequestBody");
Object.defineProperty(exports, "ConfigurationRequestBody", { enumerable: true, get: function () { return ConfigurationRequestBody_1.ConfigurationRequestBody; } });
const ParsingOption_1 = require("./ParsingOption");
Object.defineProperty(exports, "ParsingOption", { enumerable: true, get: function () { return ParsingOption_1.ParsingOption; } });
const ParsingOptionScale_1 = require("./ParsingOptionScale");
Object.defineProperty(exports, "ParsingOptionScale", { enumerable: true, get: function () { return ParsingOptionScale_1.ParsingOptionScale; } });
const Type_1 = require("./Type");
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return Type_1.Type; } });
const Chart_1 = require("./Chart");
Object.defineProperty(exports, "Chart", { enumerable: true, get: function () { return Chart_1.Chart; } });
const ChartDataSet_1 = require("./ChartDataSet");
Object.defineProperty(exports, "ChartDataSet", { enumerable: true, get: function () { return ChartDataSet_1.ChartDataSet; } });
const HoriziontalAlign_1 = require("./HoriziontalAlign");
Object.defineProperty(exports, "HoriziontalAlign", { enumerable: true, get: function () { return HoriziontalAlign_1.HoriziontalAlign; } });
const TextAlign_1 = require("./TextAlign");
Object.defineProperty(exports, "TextAlign", { enumerable: true, get: function () { return TextAlign_1.TextAlign; } });
const TitlePosition_1 = require("./TitlePosition");
Object.defineProperty(exports, "TitlePosition", { enumerable: true, get: function () { return TitlePosition_1.TitlePosition; } });
const Clip_1 = require("./Clip");
Object.defineProperty(exports, "Clip", { enumerable: true, get: function () { return Clip_1.Clip; } });
const Title_1 = require("./Title");
Object.defineProperty(exports, "Title", { enumerable: true, get: function () { return Title_1.Title; } });
const Axis_1 = require("./Axis");
Object.defineProperty(exports, "Axis", { enumerable: true, get: function () { return Axis_1.Axis; } });
const Font_1 = require("./Font");
Object.defineProperty(exports, "Font", { enumerable: true, get: function () { return Font_1.Font; } });
const Padding_1 = require("./Padding");
Object.defineProperty(exports, "Padding", { enumerable: true, get: function () { return Padding_1.Padding; } });
const Layout_1 = require("./Layout");
Object.defineProperty(exports, "Layout", { enumerable: true, get: function () { return Layout_1.Layout; } });
const Legend_1 = require("./Legend");
Object.defineProperty(exports, "Legend", { enumerable: true, get: function () { return Legend_1.Legend; } });
const LegendPosition_1 = require("./LegendPosition");
Object.defineProperty(exports, "LegendPosition", { enumerable: true, get: function () { return LegendPosition_1.LegendPosition; } });
const LegendLabel_1 = require("./LegendLabel");
Object.defineProperty(exports, "LegendLabel", { enumerable: true, get: function () { return LegendLabel_1.LegendLabel; } });
const Tooltip_1 = require("./Tooltip");
Object.defineProperty(exports, "Tooltip", { enumerable: true, get: function () { return Tooltip_1.Tooltip; } });
const PointStyle_1 = require("./PointStyle");
Object.defineProperty(exports, "PointStyle", { enumerable: true, get: function () { return PointStyle_1.PointStyle; } });
const TooltipMode_1 = require("./TooltipMode");
Object.defineProperty(exports, "TooltipMode", { enumerable: true, get: function () { return TooltipMode_1.TooltipMode; } });
const TooltipPositionMode_1 = require("./TooltipPositionMode");
Object.defineProperty(exports, "TooltipPositionMode", { enumerable: true, get: function () { return TooltipPositionMode_1.TooltipPositionMode; } });
const FontWeight_1 = require("./FontWeight");
Object.defineProperty(exports, "FontWeight", { enumerable: true, get: function () { return FontWeight_1.FontWeight; } });
const FontStyle_1 = require("./FontStyle");
Object.defineProperty(exports, "FontStyle", { enumerable: true, get: function () { return FontStyle_1.FontStyle; } });
const AxisType_1 = require("./AxisType");
Object.defineProperty(exports, "AxisType", { enumerable: true, get: function () { return AxisType_1.AxisType; } });
const AxisBorder_1 = require("./AxisBorder");
Object.defineProperty(exports, "AxisBorder", { enumerable: true, get: function () { return AxisBorder_1.AxisBorder; } });
const AxisGrid_1 = require("./AxisGrid");
Object.defineProperty(exports, "AxisGrid", { enumerable: true, get: function () { return AxisGrid_1.AxisGrid; } });
const AxisTick_1 = require("./AxisTick");
Object.defineProperty(exports, "AxisTick", { enumerable: true, get: function () { return AxisTick_1.AxisTick; } });
const AxisTimeOptions_1 = require("./AxisTimeOptions");
Object.defineProperty(exports, "AxisTimeOptions", { enumerable: true, get: function () { return AxisTimeOptions_1.AxisTimeOptions; } });
const AxisTickMajor_1 = require("./AxisTickMajor");
Object.defineProperty(exports, "AxisTickMajor", { enumerable: true, get: function () { return AxisTickMajor_1.AxisTickMajor; } });
const AxisClassification_1 = require("./AxisClassification");
Object.defineProperty(exports, "AxisClassification", { enumerable: true, get: function () { return AxisClassification_1.AxisClassification; } });
const AxisTypeRadial_1 = require("./AxisTypeRadial");
Object.defineProperty(exports, "AxisTypeRadial", { enumerable: true, get: function () { return AxisTypeRadial_1.AxisTypeRadial; } });
const AxisBounds_1 = require("./AxisBounds");
Object.defineProperty(exports, "AxisBounds", { enumerable: true, get: function () { return AxisBounds_1.AxisBounds; } });
const TickSource_1 = require("./TickSource");
Object.defineProperty(exports, "TickSource", { enumerable: true, get: function () { return TickSource_1.TickSource; } });
const AxisPosition_1 = require("./AxisPosition");
Object.defineProperty(exports, "AxisPosition", { enumerable: true, get: function () { return AxisPosition_1.AxisPosition; } });
const AxisRadial_1 = require("./AxisRadial");
Object.defineProperty(exports, "AxisRadial", { enumerable: true, get: function () { return AxisRadial_1.AxisRadial; } });
const AxisCartesian_1 = require("./AxisCartesian");
Object.defineProperty(exports, "AxisCartesian", { enumerable: true, get: function () { return AxisCartesian_1.AxisCartesian; } });
const AxisRadialAngleLine_1 = require("./AxisRadialAngleLine");
Object.defineProperty(exports, "AxisRadialAngleLine", { enumerable: true, get: function () { return AxisRadialAngleLine_1.AxisRadialAngleLine; } });
const AxisRadialPointLabel_1 = require("./AxisRadialPointLabel");
Object.defineProperty(exports, "AxisRadialPointLabel", { enumerable: true, get: function () { return AxisRadialPointLabel_1.AxisRadialPointLabel; } });
const Label_1 = require("./Label");
Object.defineProperty(exports, "Label", { enumerable: true, get: function () { return Label_1.Label; } });
const Configuration_1 = require("./Configuration");
Object.defineProperty(exports, "Configuration", { enumerable: true, get: function () { return Configuration_1.Configuration; } });
const ConfigurationData_1 = require("./ConfigurationData");
Object.defineProperty(exports, "ConfigurationData", { enumerable: true, get: function () { return ConfigurationData_1.ConfigurationData; } });
const ConfigurationDataSet_1 = require("./ConfigurationDataSet");
Object.defineProperty(exports, "ConfigurationDataSet", { enumerable: true, get: function () { return ConfigurationDataSet_1.ConfigurationDataSet; } });
const ConfigurationOptions_1 = require("./ConfigurationOptions");
Object.defineProperty(exports, "ConfigurationOptions", { enumerable: true, get: function () { return ConfigurationOptions_1.ConfigurationOptions; } });
const DataRequestBody_1 = require("./DataRequestBody");
Object.defineProperty(exports, "DataRequestBody", { enumerable: true, get: function () { return DataRequestBody_1.DataRequestBody; } });
const SourceType_1 = require("./SourceType");
Object.defineProperty(exports, "SourceType", { enumerable: true, get: function () { return SourceType_1.SourceType; } });
const ExportRequestBody_1 = require("./ExportRequestBody");
Object.defineProperty(exports, "ExportRequestBody", { enumerable: true, get: function () { return ExportRequestBody_1.ExportRequestBody; } });
const RegressionType_1 = require("./RegressionType");
Object.defineProperty(exports, "RegressionType", { enumerable: true, get: function () { return RegressionType_1.RegressionType; } });
const Regression_1 = require("./Regression");
Object.defineProperty(exports, "Regression", { enumerable: true, get: function () { return Regression_1.Regression; } });
const RegressionLine_1 = require("./RegressionLine");
Object.defineProperty(exports, "RegressionLine", { enumerable: true, get: function () { return RegressionLine_1.RegressionLine; } });
const ZoomPlugin_1 = require("./ZoomPlugin");
Object.defineProperty(exports, "ZoomPlugin", { enumerable: true, get: function () { return ZoomPlugin_1.ZoomPlugin; } });
const ZoomPluginPan_1 = require("./ZoomPluginPan");
Object.defineProperty(exports, "ZoomPluginPan", { enumerable: true, get: function () { return ZoomPluginPan_1.ZoomPluginPan; } });
const ZoomPluginMode_1 = require("./ZoomPluginMode");
Object.defineProperty(exports, "ZoomPluginMode", { enumerable: true, get: function () { return ZoomPluginMode_1.ZoomPluginMode; } });
const ZoomPluginModifierKey_1 = require("./ZoomPluginModifierKey");
Object.defineProperty(exports, "ZoomPluginModifierKey", { enumerable: true, get: function () { return ZoomPluginModifierKey_1.ZoomPluginModifierKey; } });
const ZoomPluginScaleMode_1 = require("./ZoomPluginScaleMode");
Object.defineProperty(exports, "ZoomPluginScaleMode", { enumerable: true, get: function () { return ZoomPluginScaleMode_1.ZoomPluginScaleMode; } });
const ZoomPluginOverScaleMode_1 = require("./ZoomPluginOverScaleMode");
Object.defineProperty(exports, "ZoomPluginOverScaleMode", { enumerable: true, get: function () { return ZoomPluginOverScaleMode_1.ZoomPluginOverScaleMode; } });
const ZoomPluginZoom_1 = require("./ZoomPluginZoom");
Object.defineProperty(exports, "ZoomPluginZoom", { enumerable: true, get: function () { return ZoomPluginZoom_1.ZoomPluginZoom; } });
const ZoomPluginZoomWheel_1 = require("./ZoomPluginZoomWheel");
Object.defineProperty(exports, "ZoomPluginZoomWheel", { enumerable: true, get: function () { return ZoomPluginZoomWheel_1.ZoomPluginZoomWheel; } });
const ZoomPluginZoomDrag_1 = require("./ZoomPluginZoomDrag");
Object.defineProperty(exports, "ZoomPluginZoomDrag", { enumerable: true, get: function () { return ZoomPluginZoomDrag_1.ZoomPluginZoomDrag; } });
const ZoomPluginZoomPinch_1 = require("./ZoomPluginZoomPinch");
Object.defineProperty(exports, "ZoomPluginZoomPinch", { enumerable: true, get: function () { return ZoomPluginZoomPinch_1.ZoomPluginZoomPinch; } });
const ZoomPluginScaleLimits_1 = require("./ZoomPluginScaleLimits");
Object.defineProperty(exports, "ZoomPluginScaleLimits", { enumerable: true, get: function () { return ZoomPluginScaleLimits_1.ZoomPluginScaleLimits; } });
const ZoomPluginDrawTime_1 = require("./ZoomPluginDrawTime");
Object.defineProperty(exports, "ZoomPluginDrawTime", { enumerable: true, get: function () { return ZoomPluginDrawTime_1.ZoomPluginDrawTime; } });
const AxisTime_1 = require("./AxisTime");
Object.defineProperty(exports, "AxisTime", { enumerable: true, get: function () { return AxisTime_1.AxisTime; } });
const IntervalPickerPlugin_1 = require("./IntervalPickerPlugin");
Object.defineProperty(exports, "IntervalPickerPlugin", { enumerable: true, get: function () { return IntervalPickerPlugin_1.IntervalPickerPlugin; } });
const Intervals_1 = require("./Intervals");
Object.defineProperty(exports, "Intervals", { enumerable: true, get: function () { return Intervals_1.Intervals; } });
const TimeUnit_1 = require("./TimeUnit");
Object.defineProperty(exports, "TimeUnit", { enumerable: true, get: function () { return TimeUnit_1.TimeUnit; } });
const AxisTimeDisplayFormats_1 = require("./AxisTimeDisplayFormats");
Object.defineProperty(exports, "AxisTimeDisplayFormats", { enumerable: true, get: function () { return AxisTimeDisplayFormats_1.AxisTimeDisplayFormats; } });
