// @wago/wdx-ws-client-js/build/WDX/WS/Client/JS/Service/ChartService.js
/**
 * eDesign - Runtime - Web Socket Server Package
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChartService = void 0;
const rxjs_1 = require("rxjs");
const _1 = require(".");
const WDXSchema = __importStar(require("@wago/wdx-schema-base"));
class ChartService extends _1.AbstractAPIService {
    delete(uuid) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.DeleteRequest(uuid);
        const response = new rxjs_1.Subject();
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if (WDXSchema.WDX.Schema.Message.Type.ChartDeleteResponse ===
                message.type &&
                message.uuid === request.uuid) {
                message.error ? response.error(message.error) :
                    response.next(message.body);
                response.complete();
                subscription.unsubscribe();
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
    detail(uuid) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.DetailRequest(uuid);
        const response = new rxjs_1.Subject();
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if (WDXSchema.WDX.Schema.Message.Type.ChartDetailResponse ===
                message.type &&
                message.uuid === request.uuid) {
                message.error ? response.error(message.error) :
                    response.next(message.body);
                response.complete();
                subscription.unsubscribe();
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
    save(item) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.SaveRequest(item);
        const response = new rxjs_1.Subject();
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if (message.type ===
                WDXSchema.WDX.Schema.Message.Type.ChartSaveResponse &&
                message.uuid === request.uuid) {
                message.error ? response.error(message.error) :
                    response.next(message.body);
                response.complete();
                subscription.unsubscribe();
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
    list(offset = WDXSchema.WDX.Schema.Model.Pagination.Request.DEFAULT_OFFSET, limit = WDXSchema.WDX.Schema.Model.Pagination.Request.DEFAULT_LIMIT) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.ListRequest({
            where: {},
            take: limit,
            skip: offset,
        });
        const response = new rxjs_1.Subject();
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if (message.type ===
                WDXSchema.WDX.Schema.Message.Type.ChartListResponse &&
                message.uuid === request.uuid) {
                message.error ? response.error(message.error) :
                    response.next(message.body);
                response.complete();
                subscription.unsubscribe();
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
    unregister(
    /**
     * Chart uuid
     */
    uuid) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.UnsubscribeRequest(uuid);
        const response = new rxjs_1.Subject();
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if (message.type ===
                WDXSchema.WDX.Schema.Message.Type
                    .ChartUnsubscribeResponse &&
                message.uuid === request.uuid) {
                message.error ? response.error(message.error) :
                    response.next(undefined);
                response.complete();
                subscription.unsubscribe();
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
    register(
    /**
     * Chart uuid
     */
    uuid) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.SubscribeRequest(uuid);
        const response = new rxjs_1.Subject();
        const topic = `${WDXSchema.WDX.Schema.Message.Type.ChartUpdate}-${uuid}`;
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if ((message.type ===
                WDXSchema.WDX.Schema.Message.Type
                    .ChartSubscribeResponse &&
                message.uuid === request.uuid) ||
                (topic === message.topic)) {
                message.error ? response.error(message.error) :
                    response.next(message.body);
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
    configuration(uuid) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.ConfigurationRequest(uuid);
        const response = new rxjs_1.Subject();
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if (message.type ===
                WDXSchema.WDX.Schema.Message.Type
                    .ChartConfigurationResponse &&
                message.uuid === request.uuid) {
                message.error ? response.error(message.error) :
                    response.next(message.body);
                subscription.unsubscribe();
                response.complete();
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
    data(
    /**
     * Chart uuid
     */
    uuid, dateFrom, dateTo) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.DataRequest(new WDXSchema.WDX.Schema.Model.Chart.DataRequestBody(uuid, dateFrom, dateTo));
        const response = new rxjs_1.Subject();
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if (message.type ===
                WDXSchema.WDX.Schema.Message.Type.ChartDataResponse &&
                message.uuid === request.uuid) {
                message.error ? response.error(message.error) :
                    response.next(message.body);
                subscription.unsubscribe();
                response.complete();
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
    export(uuid, type, dateFrom, dateTo) {
        const request = new WDXSchema.WDX.Schema.Message.Chart.ExportRequest(new WDXSchema.WDX.Schema.Model.Chart.ExportRequestBody(uuid, type, dateFrom, dateTo));
        const response = new rxjs_1.Subject();
        const subscription = this._clientService.incommingMessages.subscribe((message) => {
            if (message.type ===
                WDXSchema.WDX.Schema.Message.Type.ChartExportResponse &&
                message.uuid === request.uuid) {
                message.error ? response.error(message.error) :
                    response.next(message.body);
                subscription.unsubscribe();
                response.complete();
            }
        });
        this._clientService.sendMessage(request);
        return response.asObservable();
    }
}
exports.ChartService = ChartService;
