// @wago/wdx-ws-client-js/build/WDX/WS/Client/JS/Service/index.js
/**
 * eDesign - WDX - WS - Client - JS - Service Package
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChartService = exports.RuntimeService = exports.InstanceService = exports.TrendService = exports.AlarmService = exports.ScriptService = exports.DataService = exports.ClientService = exports.Status = exports.AbstractAPIService = void 0;
const AbstractAPIService_1 = require("./AbstractAPIService");
Object.defineProperty(exports, "AbstractAPIService", { enumerable: true, get: function () { return AbstractAPIService_1.AbstractAPIService; } });
const ClientService_1 = require("./ClientService");
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return ClientService_1.Status; } });
Object.defineProperty(exports, "ClientService", { enumerable: true, get: function () { return ClientService_1.ClientService; } });
const DataService_1 = require("./DataService");
Object.defineProperty(exports, "DataService", { enumerable: true, get: function () { return DataService_1.DataService; } });
const ScriptService_1 = require("./ScriptService");
Object.defineProperty(exports, "ScriptService", { enumerable: true, get: function () { return ScriptService_1.ScriptService; } });
const AlarmService_1 = require("./AlarmService");
Object.defineProperty(exports, "AlarmService", { enumerable: true, get: function () { return AlarmService_1.AlarmService; } });
const TrendService_1 = require("./TrendService");
Object.defineProperty(exports, "TrendService", { enumerable: true, get: function () { return TrendService_1.TrendService; } });
const InstanceService_1 = require("./InstanceService");
Object.defineProperty(exports, "InstanceService", { enumerable: true, get: function () { return InstanceService_1.InstanceService; } });
const RuntimeService_1 = require("./RuntimeService");
Object.defineProperty(exports, "RuntimeService", { enumerable: true, get: function () { return RuntimeService_1.RuntimeService; } });
const ChartService_1 = require("./ChartService");
Object.defineProperty(exports, "ChartService", { enumerable: true, get: function () { return ChartService_1.ChartService; } });
