// @wago/wdx-schema-base/build/WDX/Schema/Message/Script/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Messages Script
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.UploadResponse = exports.UploadRequest = exports.SaveResponse = exports.SaveRequest = exports.InstallResponse = exports.InstallRequest = exports.DeleteResponse = exports.DeleteRequest = exports.BrowseResponse = exports.BrowseRequest = void 0;
const BrowseRequest_1 = require("./BrowseRequest");
Object.defineProperty(exports, "BrowseRequest", { enumerable: true, get: function () { return BrowseRequest_1.BrowseRequest; } });
const BrowseResponse_1 = require("./BrowseResponse");
Object.defineProperty(exports, "BrowseResponse", { enumerable: true, get: function () { return BrowseResponse_1.BrowseResponse; } });
const DeleteRequest_1 = require("./DeleteRequest");
Object.defineProperty(exports, "DeleteRequest", { enumerable: true, get: function () { return DeleteRequest_1.DeleteRequest; } });
const DeleteResponse_1 = require("./DeleteResponse");
Object.defineProperty(exports, "DeleteResponse", { enumerable: true, get: function () { return DeleteResponse_1.DeleteResponse; } });
const InstallRequest_1 = require("./InstallRequest");
Object.defineProperty(exports, "InstallRequest", { enumerable: true, get: function () { return InstallRequest_1.InstallRequest; } });
const InstallResponse_1 = require("./InstallResponse");
Object.defineProperty(exports, "InstallResponse", { enumerable: true, get: function () { return InstallResponse_1.InstallResponse; } });
const SaveRequest_1 = require("./SaveRequest");
Object.defineProperty(exports, "SaveRequest", { enumerable: true, get: function () { return SaveRequest_1.SaveRequest; } });
const SaveResponse_1 = require("./SaveResponse");
Object.defineProperty(exports, "SaveResponse", { enumerable: true, get: function () { return SaveResponse_1.SaveResponse; } });
const UploadRequest_1 = require("./UploadRequest");
Object.defineProperty(exports, "UploadRequest", { enumerable: true, get: function () { return UploadRequest_1.UploadRequest; } });
const UploadResponse_1 = require("./UploadResponse");
Object.defineProperty(exports, "UploadResponse", { enumerable: true, get: function () { return UploadResponse_1.UploadResponse; } });
