// @wago/wdx-schema-base/build/WDX/Schema/Message/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Messages
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Access = exports.Alarm = exports.Chart = exports.Trend = exports.Script = exports.Runtime = exports.Type = exports.Instance = exports.Data = exports.KeepAlive = exports.AbstractMessage = exports.MESSAGE_SEPARATOR_LENGTH = exports.MESSAGE_SEPARATOR = void 0;
const Separator_1 = require("./Separator");
Object.defineProperty(exports, "MESSAGE_SEPARATOR", { enumerable: true, get: function () { return Separator_1.MESSAGE_SEPARATOR; } });
const Separator_2 = require("./Separator");
Object.defineProperty(exports, "MESSAGE_SEPARATOR_LENGTH", { enumerable: true, get: function () { return Separator_2.MESSAGE_SEPARATOR_LENGTH; } });
const AbstractMessage_1 = require("./AbstractMessage");
Object.defineProperty(exports, "AbstractMessage", { enumerable: true, get: function () { return AbstractMessage_1.AbstractMessage; } });
const KeepAlive_1 = require("./KeepAlive");
Object.defineProperty(exports, "KeepAlive", { enumerable: true, get: function () { return KeepAlive_1.KeepAlive; } });
const Type_1 = require("./Type");
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return Type_1.Type; } });
const Data = __importStar(require("./Data"));
exports.Data = Data;
const Instance = __importStar(require("./Instance"));
exports.Instance = Instance;
const Runtime = __importStar(require("./Runtime"));
exports.Runtime = Runtime;
const Script = __importStar(require("./Script"));
exports.Script = Script;
const Trend = __importStar(require("./Trend"));
exports.Trend = Trend;
const Chart = __importStar(require("./Chart"));
exports.Chart = Chart;
const Alarm = __importStar(require("./Alarm"));
exports.Alarm = Alarm;
const Access = __importStar(require("./Access"));
exports.Access = Access;
