// @wago/wdx-schema-base/build/WDX/Schema/Message/Type.js
/**
 * Elrest eDesign Runtime Library Messages
 *
 * @copyright 2024 Elrest AutomationsSysteme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = void 0;
var Type;
(function (Type) {
    /**
     * Data
     */
    Type["DataUpdate"] = "DataUpdate";
    Type["DataRefreshSchemaRequest"] = "DataRefreshSchemaRequest";
    Type["DataRefreshSchemaResponse"] = "DataRefreshSchemaResponse";
    Type["DataRegisterProviderRequest"] = "DataRegisterProviderRequest";
    Type["DataRegisterProviderResponse"] = "DataRegisterProviderResponse";
    Type["DataProviderRequest"] = "DataProviderRequest";
    Type["DataProviderResponse"] = "DataProviderResponse";
    Type["DataRegisterValueRequest"] = "DataRegisterValueRequest";
    Type["DataRegisterValueResponse"] = "DataRegisterValueResponse";
    Type["DataUnregisterValueRequest"] = "DataUnregisterValueRequest";
    Type["DataUnregisterValueResponse"] = "DataUnregisterValueResponse";
    Type["DataRegisterSchemaChangesRequest"] = "DataRegisterSchemaChangesRequest";
    Type["DataRegisterSchemaChangesResponse"] = "DataRegisterSchemaChangesResponse";
    Type["DataUnregisterSchemaChangesRequest"] = "DataUnregisterSchemaChangesRequest";
    Type["DataUnregisterSchemaChangesResponse"] = "DataUnregisterSchemaChangesResponse";
    Type["DataSchemaChanges"] = "DataSchemaChanges";
    Type["DataSetValueRequest"] = "DataSetValueRequest";
    Type["DataSetValueResponse"] = "DataSetValueResponse";
    Type["DataGetValueRequest"] = "DataGetValueRequest";
    Type["DataGetValueResponse"] = "DataGetValueResponse";
    Type["DataSetSchemaRequest"] = "DataSetSchemaRequest";
    Type["DataSetSchemaResponse"] = "DataSetSchemaResponse";
    Type["DataGetSchemaRequest"] = "DataGetSchemaRequest";
    Type["DataGetSchemaResponse"] = "DataGetSchemaResponse";
    Type["DataDeleteSchemaRequest"] = "DataDeleteSchemaRequest";
    Type["DataDeleteSchemaResponse"] = "DataDeleteSchemaResponse";
    /**
     * Instance
     */
    Type["InstanceDeleteRequest"] = "InstanceDeleteRequest";
    Type["InstanceDeleteResponse"] = "InstanceDeleteResponse";
    Type["InstanceIAm"] = "InstanceIAm";
    Type["InstanceIAmError"] = "InstanceIAmError";
    Type["InstanceIAmGone"] = "InstanceIAmGone";
    Type["InstanceDetailRequest"] = "InstanceDetailRequest";
    Type["InstanceDetailResponse"] = "InstanceDetailResponse";
    Type["InstanceListRequest"] = "InstanceListRequest";
    Type["InstanceListResponse"] = "InstanceListResponse";
    Type["InstanceListNamespacesRequest"] = "InstanceListNamespacesRequest";
    Type["InstanceListNamespacesResponse"] = "InstanceListNamespacesResponse";
    /** Instance Logs */
    Type["InstanceLog"] = "InstanceLog";
    Type["InstanceListLogRequest"] = "InstanceListLogRequest";
    Type["InstanceListLogResponse"] = "InstanceListLogResponse";
    Type["InstanceLogSubscribeRequest"] = "InstanceLogSubscribeRequest";
    Type["InstanceLogSubscribeResponse"] = "InstanceLogSubscribeResponse";
    Type["InstanceLogUnsubscribeRequest"] = "InstanceLogUnsubscribeRequest";
    Type["InstanceLogUnsubscribeResponse"] = "InstanceLogUnsubscribeResponse";
    /** Instance CRUD */
    Type["InstanceRestartRequest"] = "InstanceRestartRequest";
    Type["InstanceRestartResponse"] = "InstanceRestartResponse";
    Type["InstanceStartRequest"] = "InstanceStartRequest";
    Type["InstanceStartResponse"] = "InstanceStartResponse";
    Type["InstanceStopRequest"] = "InstanceStopRequest";
    Type["InstanceStopResponse"] = "InstanceStopResponse";
    Type["InstanceSaveRequest"] = "InstanceSaveRequest";
    Type["InstanceSaveResponse"] = "InstanceSaveResponse";
    Type["InstanceWhoIsRequest"] = "InstanceWhoIsRequest";
    Type["InstanceWhoIsResponse"] = "InstanceWhoIsResponse";
    Type["InstanceMonitor"] = "InstanceMonitor";
    Type["InstanceMonitorSubscribeRequest"] = "InstanceMonitorSubscribeRequest";
    Type["InstanceMonitorSubscribeResponse"] = "InstanceMonitorSubscribeResponse";
    Type["InstanceMonitorUnsubscribeRequest"] = "InstanceMonitorUnsubscribeRequest";
    Type["InstanceMonitorUnsubscribeResponse"] = "InstanceMonitorUnsubscribeResponse";
    /**
     * Package
     */
    Type["PackageInstallRequest"] = "PackageInstallRequest";
    Type["PackageInstallResponse"] = "PackageInstallResponse";
    Type["PackageUninstallRequest"] = "PackageUninstallRequest";
    Type["PackageUninstallResponse"] = "PackageUninstallResponse";
    /**
     * Repository
     */
    Type["RepositoryListRequest"] = "RepositoryListRequest";
    Type["RepositoryListResponse"] = "RepositoryListResponse";
    Type["RepositoryInfoRequest"] = "RepositoryInfoRequest";
    Type["RepositoryInfoResponse"] = "RepositoryInfoResponse";
    /**
     * Runtime
     */
    Type["RuntimeMonitor"] = "RuntimeMonitor";
    Type["RuntimeLogrotateRequest"] = "RuntimeLogrotateRequest";
    Type["RuntimeLogrotateResponse"] = "RuntimeLogrotateResponse";
    Type["RuntimeInfoRequest"] = "RuntimeInfoRequest";
    Type["RuntimeInfoResponse"] = "RuntimeInfoResponse";
    Type["RuntimeMonitorSubscribeRequest"] = "RuntimeMonitorSubscribeRequest";
    Type["RuntimeMonitorSubscribeResponse"] = "RuntimeMonitorSubscribeResponse";
    Type["RuntimeMonitorUnsubscribeRequest"] = "RuntimeMonitorUnsubscribeRequest";
    Type["RuntimeMonitorUnsubscribeResponse"] = "RuntimeMonitorUnsubscribeResponse";
    Type["RuntimeStartRequest"] = "RuntimeStartRequest";
    Type["RuntimeStartResponse"] = "RuntimeStartResponse";
    Type["RuntimeStopRequest"] = "RuntimeStopRequest";
    Type["RuntimeStopResponse"] = "RuntimeStopResponse";
    Type["RuntimeRestartRequest"] = "RuntimeRestartRequest";
    Type["RuntimeRestartResponse"] = "RuntimeRestartResponse";
    /**
     * JS Workspace
     */
    Type["ScriptBrowseRequest"] = "ScriptBrowseRequest";
    Type["ScriptBrowseResponse"] = "ScriptBrowseResponse";
    Type["ScriptInstallRequest"] = "ScriptInstallRequest";
    Type["ScriptInstallResponse"] = "ScriptInstallResponse";
    Type["ScriptSaveRequest"] = "ScriptSaveRequest";
    Type["ScriptSaveResponse"] = "ScriptSaveResponse";
    Type["ScriptDeleteRequest"] = "ScriptDeleteRequest";
    Type["ScriptDeleteResponse"] = "ScriptDeleteResponse";
    Type["ScriptUploadRequest"] = "ScriptUploadRequest";
    Type["ScriptUploadResponse"] = "ScriptUploadResponse";
    Type["ScriptDownloadRequest"] = "ScriptDownloadRequest";
    Type["ScriptDownloadResponse"] = "ScriptDownloadResponse";
    /**
     * KeepALive
     */
    Type["KeepALive"] = "KeepAlive";
    /**
     * Trending
     */
    Type["TrendingListRequest"] = "TrendingListRequest";
    Type["TrendingListResponse"] = "TrendingListResponse";
    Type["TrendingDetailRequest"] = "TrendingDetailRequest";
    Type["TrendingDetailResponse"] = "TrendingDetailResponse";
    Type["TrendingSaveRequest"] = "TrendingSaveRequest";
    Type["TrendingSaveResponse"] = "TrendingSaveResponse";
    Type["TrendingDeleteRequest"] = "TrendingDeleteRequest";
    Type["TrendingDeleteResponse"] = "TrendingDeleteResponse";
    Type["TrendingSubscribeRequest"] = "TrendingSubscribeRequest";
    Type["TrendingSubscribeResponse"] = "TrendingSubscribeResponse";
    Type["TrendingUnsubscribeRequest"] = "TrendingUnsubscribeRequest";
    Type["TrendingUnsubscribeResponse"] = "TrendingUnsubscribeResponse";
    Type["TrendingUpdate"] = "TrendingUpdate";
    Type["TrendingDataRequest"] = "TrendingDataRequest";
    Type["TrendingDataResponse"] = "TrendingDataResponse";
    Type["TrendingExportRequest"] = "TrendingExportRequest";
    Type["TrendingExportResponse"] = "TrendingExportResponse";
    /**
     * Charts
     */
    Type["ChartConfigurationRequest"] = "ChartConfigurationRequest";
    Type["ChartConfigurationResponse"] = "ChartConfigurationResponse";
    Type["ChartDataRequest"] = "ChartDataRequest";
    Type["ChartDataResponse"] = "ChartDataResponse";
    Type["ChartDetailRequest"] = "ChartDetailRequest";
    Type["ChartDetailResponse"] = "ChartDetailResponse";
    Type["ChartDeleteRequest"] = "ChartDeleteRequest";
    Type["ChartDeleteResponse"] = "ChartDeleteResponse";
    Type["ChartExportRequest"] = "ChartExportRequest";
    Type["ChartExportResponse"] = "ChartExportResponse";
    Type["ChartListRequest"] = "ChartListRequest";
    Type["ChartListResponse"] = "ChartListResponse";
    Type["ChartSaveRequest"] = "ChartSaveRequest";
    Type["ChartSaveResponse"] = "ChartSaveResponse";
    Type["ChartSubscribeRequest"] = "ChartSubscribeRequest";
    Type["ChartSubscribeResponse"] = "ChartSubscribeResponse";
    Type["ChartUpdate"] = "ChartUpdate";
    Type["ChartUnsubscribeRequest"] = "ChartUnsubscribeRequest";
    Type["ChartUnsubscribeResponse"] = "ChartUnsubscribeResponse";
    /**
     * Alarming
     */
    Type["AlarmingListRequest"] = "AlarmingListRequest";
    Type["AlarmingListResponse"] = "AlarmingListResponse";
    Type["AlarmingDetailRequest"] = "AlarmingDetailRequest";
    Type["AlarmingDetailResponse"] = "AlarmingDetailResponse";
    Type["AlarmingListHistoryRequest"] = "AlarmingListHistoryRequest";
    Type["AlarmingListHistoryResponse"] = "AlarmingListHistoryResponse";
    Type["AlarmingExportHistoryRequest"] = "AlarmingExportHistoryRequest";
    Type["AlarmingExportHistoryResponse"] = "AlarmingExportHistoryResponse";
    Type["AlarmingDeleteHistoryRequest"] = "AlarmingDeleteHistoryRequest";
    Type["AlarmingDeleteHistoryResponse"] = "AlarmingDeleteHistoryResponse";
    Type["AlarmingExportRequest"] = "AlarmingExportRequest";
    Type["AlarmingExportResponse"] = "AlarmingExportResponse";
    Type["AlarmingSetRequest"] = "AlarmingSetRequest";
    Type["AlarmingSetResponse"] = "AlarmingSetResponse";
    Type["AlarmingConfirmRequest"] = "AlarmingConfirmRequest";
    Type["AlarmingConfirmResponse"] = "AlarmingConfirmResponse";
    Type["AlarmingDeleteRequest"] = "AlarmingDeleteRequest";
    Type["AlarmingDeleteResponse"] = "AlarmingDeleteResponse";
    Type["AlarmingSubscribeRequest"] = "AlarmingSubscribeRequest";
    Type["AlarmingSubscribeResponse"] = "AlarmingSubscribeResponse";
    Type["AlarmingUnsubscribeRequest"] = "AlarmingUnsubscribeRequest";
    Type["AlarmingUnsubscribeResponse"] = "AlarmingUnsubscribeResponse";
    Type["AlarmingGetStatusRequest"] = "AlarmingGetStatusRequest";
    Type["AlarmingGetStatusResponse"] = "AlarmingGetStatusResponse";
    Type["AlarmingSetActiveRequest"] = "AlarmingSetActiveRequest";
    Type["AlarmingSetActiveResponse"] = "AlarmingSetActiveResponse";
    Type["AlarmingSetInactiveRequest"] = "AlarmingSetInactiveRequest";
    Type["AlarmingSetInactiveResponse"] = "AlarmingSetInactiveResponse";
    Type["AlarmingUpdate"] = "AlarmingUpdate";
    Type["AuthLogin"] = "AuthLogin";
    Type["AuthLogout"] = "AuthLogout";
    Type["AccessUserListRequest"] = "AccessUserListRequest";
    Type["AccessUserListResponse"] = "AccessUserListResponse";
    Type["AccessUserSaveRequest"] = "AccessUserSaveRequest";
    Type["AccessUserSaveResponse"] = "AccessUserSaveResponse";
    Type["AccessUserDeleteRequest"] = "AccessUserDeleteRequest";
    Type["AccessUserDeleteResponse"] = "AccessUserDeleteResponse";
    Type["AccessRoleListRequest"] = "AccessRoleListRequest";
    Type["AccessRoleListResponse"] = "AccessRoleListResponse";
    Type["AccessRoleSaveRequest"] = "AccessRoleSaveRequest";
    Type["AccessRoleSaveResponse"] = "AccessRoleSaveResponse";
    Type["AccessRoleDeleteRequest"] = "AccessRoleDeleteRequest";
    Type["AccessRoleDeleteResponse"] = "AccessRoleDeleteResponse";
    Type["AccessResourceListRequest"] = "AccessResourceListRequest";
    Type["AccessResourceListResponse"] = "AccessResourceListResponse";
})(Type || (exports.Type = Type = {}));
