// @wago/wdx-schema-base/build/WDX/Schema/Model/Script/index.js
/**
 * Elrest eDesign Runtime IPC Typescript Model Script
 *
 * @copyright 2024 Elrest Automations Systeme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.Script = exports.UploadRequestBody = exports.BrowseRequestBody = exports.INSTANCE_NAMESPACE = exports.BROWSE_NO_LEVEL = exports.BROWSE_DEFAULT_LEVEL = void 0;
exports.BROWSE_DEFAULT_LEVEL = 1;
exports.BROWSE_NO_LEVEL = -1;
exports.INSTANCE_NAMESPACE = 'JS Scripts';
const BrowseRequestBody_1 = require("./BrowseRequestBody");
Object.defineProperty(exports, "BrowseRequestBody", { enumerable: true, get: function () { return BrowseRequestBody_1.BrowseRequestBody; } });
const UploadRequestBody_1 = require("./UploadRequestBody");
Object.defineProperty(exports, "UploadRequestBody", { enumerable: true, get: function () { return UploadRequestBody_1.UploadRequestBody; } });
const Script_1 = require("./Script");
Object.defineProperty(exports, "Script", { enumerable: true, get: function () { return Script_1.Script; } });
