// @wago/wdx-schema-base/build/WDX/Schema/Model/Instance/index.js
/**
 * Elrest eDesign Runtime Library Messages Model Instance
 *
 * @copyright 2024 Elrest AutomationsSysteme GMBH
 */
'use strict';
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListLogRequestBody = exports.InstanceMonitor = exports.InstanceOrigin = exports.AlarmOptions = exports.WebSocketServerExecutionOptions = exports.NodeREDExecutionOptions = exports.JSSWorkspaceExecutionOptions = exports.JSScriptExecutionOptions = exports.DataExecutionOptions = exports.ControllerExecutionOptions = exports.AlarmExecutionOptions = exports.WebSocketServerInstance = exports.TrendInstance = exports.JSScriptInstance = exports.JSSWorkspaceInstance = exports.DataInstance = exports.ControllerInstance = exports.AlarmInstance = exports.NodeREDInstance = exports.DataSourceOptions = exports.DataAdapter = exports.Type = exports.WebSocketServerOptions = exports.UDPOptions = exports.TCPOptions = exports.Status = exports.NodeREDOptions = exports.RestartOptions = exports.LogOptions = exports.Instance = exports.ExecutionOptions = exports.ExecutionMode = exports.LogLevel = exports.ResourceUsage = exports.WSProtocol = exports.LogDate = exports.Log = exports.APIServerExecutionOptions = exports.APIServerOptions = exports.APIServerInstance = exports.AuthServerInstance = exports.AuthServerExecutionOptions = exports.AuthServerOptions = exports.BPMNInstance = exports.TrendExecutionOptions = exports.TCPOptionsListen = exports.TCPOptionsServer = exports.IPCType = void 0;
const Log_1 = require("./Log");
Object.defineProperty(exports, "Log", { enumerable: true, get: function () { return Log_1.Log; } });
const LogDate_1 = require("./LogDate");
Object.defineProperty(exports, "LogDate", { enumerable: true, get: function () { return LogDate_1.LogDate; } });
const WSProtocol_1 = require("./WSProtocol");
Object.defineProperty(exports, "WSProtocol", { enumerable: true, get: function () { return WSProtocol_1.WSProtocol; } });
const ResourceUsage_1 = require("./ResourceUsage");
Object.defineProperty(exports, "ResourceUsage", { enumerable: true, get: function () { return ResourceUsage_1.ResourceUsage; } });
const LogLevel_1 = require("./LogLevel");
Object.defineProperty(exports, "LogLevel", { enumerable: true, get: function () { return LogLevel_1.LogLevel; } });
const ExecutionMode_1 = require("./ExecutionMode");
Object.defineProperty(exports, "ExecutionMode", { enumerable: true, get: function () { return ExecutionMode_1.ExecutionMode; } });
const ExecutionOptions_1 = require("./ExecutionOptions");
Object.defineProperty(exports, "ExecutionOptions", { enumerable: true, get: function () { return ExecutionOptions_1.ExecutionOptions; } });
const Instance_1 = require("./Instance");
Object.defineProperty(exports, "Instance", { enumerable: true, get: function () { return Instance_1.Instance; } });
const LogOptions_1 = require("./LogOptions");
Object.defineProperty(exports, "LogOptions", { enumerable: true, get: function () { return LogOptions_1.LogOptions; } });
const RestartOptions_1 = require("./RestartOptions");
Object.defineProperty(exports, "RestartOptions", { enumerable: true, get: function () { return RestartOptions_1.RestartOptions; } });
const Status_1 = require("./Status");
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return Status_1.Status; } });
const NodeREDOptions_1 = require("./NodeREDOptions");
Object.defineProperty(exports, "NodeREDOptions", { enumerable: true, get: function () { return NodeREDOptions_1.NodeREDOptions; } });
const WebSocketServerOptions_1 = require("./WebSocketServerOptions");
Object.defineProperty(exports, "WebSocketServerOptions", { enumerable: true, get: function () { return WebSocketServerOptions_1.WebSocketServerOptions; } });
const DataAdapter = __importStar(require("./DataAdapter"));
exports.DataAdapter = DataAdapter;
const Type_1 = require("./Type");
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return Type_1.Type; } });
const TCPOptions_1 = require("./TCPOptions");
Object.defineProperty(exports, "TCPOptions", { enumerable: true, get: function () { return TCPOptions_1.TCPOptions; } });
const UDPOptions_1 = require("./UDPOptions");
Object.defineProperty(exports, "UDPOptions", { enumerable: true, get: function () { return UDPOptions_1.UDPOptions; } });
const DataSourceOptions_1 = require("./DataSourceOptions");
Object.defineProperty(exports, "DataSourceOptions", { enumerable: true, get: function () { return DataSourceOptions_1.DataSourceOptions; } });
const NodeREDInstance_1 = require("./NodeREDInstance");
Object.defineProperty(exports, "NodeREDInstance", { enumerable: true, get: function () { return NodeREDInstance_1.NodeREDInstance; } });
const AlarmOptions_1 = require("./AlarmOptions");
Object.defineProperty(exports, "AlarmOptions", { enumerable: true, get: function () { return AlarmOptions_1.AlarmOptions; } });
const AlarmInstance_1 = require("./AlarmInstance");
Object.defineProperty(exports, "AlarmInstance", { enumerable: true, get: function () { return AlarmInstance_1.AlarmInstance; } });
const ControllerInstance_1 = require("./ControllerInstance");
Object.defineProperty(exports, "ControllerInstance", { enumerable: true, get: function () { return ControllerInstance_1.ControllerInstance; } });
const DataInstance_1 = require("./DataInstance");
Object.defineProperty(exports, "DataInstance", { enumerable: true, get: function () { return DataInstance_1.DataInstance; } });
const JSSWorkspaceInstance_1 = require("./JSSWorkspaceInstance");
Object.defineProperty(exports, "JSSWorkspaceInstance", { enumerable: true, get: function () { return JSSWorkspaceInstance_1.JSSWorkspaceInstance; } });
const JSScriptInstance_1 = require("./JSScriptInstance");
Object.defineProperty(exports, "JSScriptInstance", { enumerable: true, get: function () { return JSScriptInstance_1.JSScriptInstance; } });
const TrendInstance_1 = require("./TrendInstance");
Object.defineProperty(exports, "TrendInstance", { enumerable: true, get: function () { return TrendInstance_1.TrendInstance; } });
const WebSocketServerInstance_1 = require("./WebSocketServerInstance");
Object.defineProperty(exports, "WebSocketServerInstance", { enumerable: true, get: function () { return WebSocketServerInstance_1.WebSocketServerInstance; } });
const AlarmExecutionOptions_1 = require("./AlarmExecutionOptions");
Object.defineProperty(exports, "AlarmExecutionOptions", { enumerable: true, get: function () { return AlarmExecutionOptions_1.AlarmExecutionOptions; } });
const ControllerExecutionOptions_1 = require("./ControllerExecutionOptions");
Object.defineProperty(exports, "ControllerExecutionOptions", { enumerable: true, get: function () { return ControllerExecutionOptions_1.ControllerExecutionOptions; } });
const DataExecutionOptions_1 = require("./DataExecutionOptions");
Object.defineProperty(exports, "DataExecutionOptions", { enumerable: true, get: function () { return DataExecutionOptions_1.DataExecutionOptions; } });
const JSScriptExecutionOptions_1 = require("./JSScriptExecutionOptions");
Object.defineProperty(exports, "JSScriptExecutionOptions", { enumerable: true, get: function () { return JSScriptExecutionOptions_1.JSScriptExecutionOptions; } });
const JSSWorkspaceExecutionOptions_1 = require("./JSSWorkspaceExecutionOptions");
Object.defineProperty(exports, "JSSWorkspaceExecutionOptions", { enumerable: true, get: function () { return JSSWorkspaceExecutionOptions_1.JSSWorkspaceExecutionOptions; } });
const NodeREDExecutionOptions_1 = require("./NodeREDExecutionOptions");
Object.defineProperty(exports, "NodeREDExecutionOptions", { enumerable: true, get: function () { return NodeREDExecutionOptions_1.NodeREDExecutionOptions; } });
const WebSocketServerExecutionOptions_1 = require("./WebSocketServerExecutionOptions");
Object.defineProperty(exports, "WebSocketServerExecutionOptions", { enumerable: true, get: function () { return WebSocketServerExecutionOptions_1.WebSocketServerExecutionOptions; } });
const InstanceOrigin_1 = require("./InstanceOrigin");
Object.defineProperty(exports, "InstanceOrigin", { enumerable: true, get: function () { return InstanceOrigin_1.InstanceOrigin; } });
const APIServerInstance_1 = require("./APIServerInstance");
Object.defineProperty(exports, "APIServerInstance", { enumerable: true, get: function () { return APIServerInstance_1.APIServerInstance; } });
const APIServerOptions_1 = require("./APIServerOptions");
Object.defineProperty(exports, "APIServerOptions", { enumerable: true, get: function () { return APIServerOptions_1.APIServerOptions; } });
const APIServerExecutionOptions_1 = require("./APIServerExecutionOptions");
Object.defineProperty(exports, "APIServerExecutionOptions", { enumerable: true, get: function () { return APIServerExecutionOptions_1.APIServerExecutionOptions; } });
const InstanceMonitor_1 = require("./InstanceMonitor");
Object.defineProperty(exports, "InstanceMonitor", { enumerable: true, get: function () { return InstanceMonitor_1.InstanceMonitor; } });
const AuthServerOptions_1 = require("./AuthServerOptions");
Object.defineProperty(exports, "AuthServerOptions", { enumerable: true, get: function () { return AuthServerOptions_1.AuthServerOptions; } });
const AuthServerExecutionOptions_1 = require("./AuthServerExecutionOptions");
Object.defineProperty(exports, "AuthServerExecutionOptions", { enumerable: true, get: function () { return AuthServerExecutionOptions_1.AuthServerExecutionOptions; } });
const AuthServerInstance_1 = require("./AuthServerInstance");
Object.defineProperty(exports, "AuthServerInstance", { enumerable: true, get: function () { return AuthServerInstance_1.AuthServerInstance; } });
const IPCType_1 = require("./IPCType");
Object.defineProperty(exports, "IPCType", { enumerable: true, get: function () { return IPCType_1.IPCType; } });
const TCPOptionsServer_1 = require("./TCPOptionsServer");
Object.defineProperty(exports, "TCPOptionsServer", { enumerable: true, get: function () { return TCPOptionsServer_1.TCPOptionsServer; } });
const TCPOptionsListen_1 = require("./TCPOptionsListen");
Object.defineProperty(exports, "TCPOptionsListen", { enumerable: true, get: function () { return TCPOptionsListen_1.TCPOptionsListen; } });
const BPMNInstance_1 = require("./BPMNInstance");
Object.defineProperty(exports, "BPMNInstance", { enumerable: true, get: function () { return BPMNInstance_1.BPMNInstance; } });
const TrendExecutionOptions_1 = require("./TrendExecutionOptions");
Object.defineProperty(exports, "TrendExecutionOptions", { enumerable: true, get: function () { return TrendExecutionOptions_1.TrendExecutionOptions; } });
const ListLogRequestBody_1 = require("./ListLogRequestBody");
Object.defineProperty(exports, "ListLogRequestBody", { enumerable: true, get: function () { return ListLogRequestBody_1.ListLogRequestBody; } });
