// @wago/wdx-schema-base/build/WDX/Schema/Model/Instance/Type.js
/**
 * Elrest eDesign Runtime Library Messages Model Instance  Module
 * Type Enumeration ServerOpts Implementation
 *
 * @copyright 2024 Elrest AutomationsSysteme GMBH
 */
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = void 0;
var Type;
(function (Type) {
    Type["AuthServer"] = "AuthServer";
    Type["Alarm"] = "Alarm";
    Type["Controller"] = "WDX";
    Type["ControllerCLI"] = "WDX-CLI";
    Type["Data"] = "Data";
    Type["JSWorkspace"] = "JSWorkspace";
    Type["JSScript"] = "JSScript";
    Type["NodeRED"] = "NodeRED";
    Type["Trend"] = "Trend";
    Type["WebSocketServer"] = "WebSocketServer";
    Type["BACNetDataAdapter"] = "BACNet";
    Type["CC100IODataAdapter"] = "CC100-IO";
    Type["eWebDataAdapter"] = "eWeb";
    Type["MODBUSDataAdapter"] = "MODBUS";
    Type["MQTTDataAdapter"] = "MQTT";
    Type["ORMDataAdapter"] = "ORM";
    Type["VirtualDataAdapter"] = "Virtual";
    Type["MQDataAdapter"] = "MQ";
    Type["OPCUADataAdapter"] = "OPCUA";
    Type["APIServer"] = "API";
    Type["BPMN"] = "BPMN";
})(Type || (exports.Type = Type = {}));
